lib/Core/StatsTracker.cpp

```diff
@@ -285,18 +286,19 @@ void StatsTracker::done() {
 void StatsTracker::stepInstruction(ExecutionState &es) {
   if (OutputIStats) {
     if (TrackInstructionTime) {
-      static sys::TimeValue lastNowTime(0,0),lastUserTime(0,0);
-    
-      if (lastUserTime.seconds()==0 && lastUserTime.nanoseconds()==0) {
-        sys::TimeValue sys(0,0);
-        sys::Process::GetTimeUsage(lastNowTime,lastUserTime,sys);
+      static time::Point lastNowTime(time::getWallTime());
+      static time::Span lastUserTime;
+
+      if (!lastUserTime) {
+        std::tie(lastUserTime, std::ignore) = time::getElapsedTime();
       } else {
-        sys::TimeValue now(0,0),user(0,0),sys(0,0);
-        sys::Process::GetTimeUsage(now,user,sys);
-        sys::TimeValue delta = user - lastUserTime;
-        sys::TimeValue deltaNow = now - lastNowTime;
-        stats::instructionTime += delta.usec();
-        stats::instructionRealTime += deltaNow.usec();
+        const auto now = time::getWallTime();
+        time::Span user, sys;
+        std::tie(user, sys) = time::getElapsedTime();
+        const auto delta = user - lastUserTime;
+        const auto deltaNow = now - lastNowTime;
+        stats::instructionTime += static_cast<std::uint64_t >(delta);
```

Comment:
I think it's confusing that this cast silently converts to microseconds. Perhaps `toMicroseconds()`?